silk-react-components/src/HierarchicalMapping/Components/MappingRule/ObjectMappingRule.jsx

```diff
@@ -29,49 +29,61 @@ import {
     MAPPING_RULE_TYPE_ROOT,
     MAPPING_RULE_TYPE_URI,
 } from '../../helpers';
+import { MESSAGES } from '../../constants';
+import transformRuleOfObjectMapping from '../../utils/transformRuleOfObjectMapping';
+import EventEmitter from '../../utils/EventEmitter';
 
-const ObjectRule = React.createClass({
-    mixins: [UseMessageBus],
-
+class ObjectRule extends React.Component {
+    state = {
+        edit: !!this.props.edit,
```

Comment:
when `edit` is `PropTypes.bool.isRequired` why typecasting is needed?

Reply:
`props.edit` can be false or Object :)
That is why typecasting still alive in there

Reply:
Okay, but when it can be Object too, the Proptype seems to be not correct